core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -351,12 +351,14 @@ class KafkaController(val config: KafkaConfig,
   }
 
   /**
-   * This callback is invoked by the replica state machine's broker change listener, with the list of newly started
+   * This callback is invoked by the broker change listener, with the list of newly started
    * brokers as input. It does the following -
    * 1. Sends update metadata request to all live and shutting down brokers
    * 2. Triggers the OnlinePartition state change for all new/offline partitions
    * 3. It checks whether there are reassigned replicas assigned to any newly started brokers. If
    *    so, it performs the reassignment logic for each topic/partition.
+   * 4. It checks whether there are replicas, whose topics are queued up for deletion, assigned to any newly started brokers.
```

Comment:
nit: maybe drop `It` from 3 and 4 to match the first two points.

The phrasing is a little wordy. How about this?
> 4. Resumes deletion of topics with replicas on the newly started brokers